src/lib.rs

```diff
@@ -126,8 +128,10 @@ impl EncodingDetector {
             match charset {
                 Err(_) =>
                     panic!("uchardet_get_charset returned invalid value"),
-                Ok("") => None,
-                Ok(encoding) => Some(encoding.to_string())
+                Ok("") => Err(EncodingDetectorError {
+                    message: "uchardet failed to recognize a charset".to_string()
```

Comment:
This looks like a good idea, I need to double-check this new behavior against `substudy` to see if it still gets good results with older subtitle files.

And if I convert the library to use error-chain, I want to break this out as a distinct error type so that our callers can handle it specially.